keg_bouncer/model/migration.py

```diff
@@ -21,3 +24,68 @@ def link_user_to_user_groups(op, user_primary_key_column):
 
 def drop_link_from_user_to_user_groups(op):
     op.drop_table(user_user_group_link_table_name)
+
+
+def alembic_create_permission(op, permissions=[{}], groups=set(), bundles=set()):
+    """Bulk create permissions and add them to groups and bundles
+
+    :param permisisons: a list of dictionaries with 'token' and 'description' keys
+    :param groups: a set of group labels, missing groups will be created
+    :param bundles: a set of bundle labels, missing bundles will be created
+    """
+
+    new_permission_tokens = set([perm['token'] for perm in permissions])
+    groups = set(groups)
+    bundles = set(bundles)
+    con = op.get_bind()
+    meta = sa.MetaData()
+    execute = con.execute
+
+    # inspect the tables we need and create a table object for them
+    permissions_table = sa.Table('keg_bouncer_permissions', meta, autoload=True,
+                                 autoload_with=con.engine)
+
+    groups_table = sa.Table('keg_bouncer_user_groups', meta, autoload=True,
+                            autoload_with=con.engine)
+
+    groups_map = sa.Table('keg_bouncer_user_group_permission_map', meta, autoload=True,
+                          autoload_with=con.engine)
+
+    bundles_table = sa.Table('keg_bouncer_permission_bundles', meta, autoload=True,
+                             autoload_with=con.engine)
+
+    bundles_map = sa.Table('keg_bouncer_bundle_permission_map', meta, autoload=True,
+                           autoload_with=con.engine)
```

Comment:
> Inspecting the table allows you to separate from the code base and use whatever is in the database right now.
But not really, if you hard-code column names. My point is, if your ORM code represents the current state of the DB*, then using the ORM will let you use this migration just fine. If your code does not represent the current state of the DB* then this migration *still* might fail, because it hard-codes column names.

\* As it relates to KegBouncer's tables.

So using the ORM lets you live in 2 possibilities:

| DB State | Migration |
|---|---|
| ORM == DB | works   |
| ORM != DB | fails  |

But inspecting the tables and then using hard-coded column names allows us to live in 4 possibilities:

| DB State | Migration |
|---|---|
| ORM == DB (and migration does) | works |
| ORM == DB (but migration does not) | fails |
| ORM != DB (and migration does) | works  |
| ORM == DB (but migration does not ) | fails  |

My point is, have 4 possibilities is more complex than just two. The both have 50% chance of failure, so we might as well use the one that's simpler.